rdflib/graph.py

```diff
@@ -952,14 +953,19 @@ def serialize(
         Format support can be extended with plugins,
         but "xml", "n3", "turtle", "nt", "pretty-xml", "trix", "trig" and "nquads" are built in.
         """
+
+        # if a base is set here, use it, else if it was set at graph creation time, use that (self.base)
+        if base is not None:
+            self.base = base
+
```

Comment:
I don't like this. It would overwrite the base as it was set on the graph.
I would expect a base that I set on the graph object to remain like that unless I explicitly change it on the graph object. If I set a base for the call to the `serialize` method I would expect this base to be only valid for the run time of this serialization.

I would make the behavior the other way round and set the local variable to `self.base`, if `base is None`.

Reply:
> I would expect this base to be only valid for the run time of this serialization.

I agree, I think this is better. I'll make this change.